dask/highlevelgraph.py

```diff
@@ -57,12 +57,40 @@ class Layer(collections.abc.Mapping):
     """
 
     annotations: Optional[Mapping[str, Any]]
+    collection_annotations: Optional[Mapping[str, Any]]
 
-    def __init__(self, annotations: Mapping[str, Any] = None):
+    def __init__(
+        self,
+        annotations: Mapping[str, Any] = None,
+        collection_annotations: Mapping[str, Any] = {},
+    ):
+        """Initialize Layer object.
+
+        Parameters
+        ----------
+        annotations : Mapping[str, Any], optional
+            By default, None.
+            Annotations are metadata or soft constraints associated with tasks
+            that dask schedulers may choose to respect:
+            They signal intent without enforcing hard constraints.
+            As such, they are primarily designed for use with the distributed
+            scheduler. See the dask.annotate function for more information.
+        collection_annotations : Mapping[str, Any], optional
+            By default an empty dict.
+            Experimental, intended to assist with visualizing the performance
+            characteristics of Dask computations.
+            These annotations are *not* passed to the distributed scheduler.
+        """
         if annotations:
             self.annotations = annotations
         else:
             self.annotations = copy.copy(config.get("annotations", None))
+        if collection_annotations:
+            self.collection_annotations = collection_annotations
+        else:
+            self.collection_annotations = copy.copy(
+                config.get("collection_annotations", {})
+            )
```

Comment:
Is the intent here to support the following?

```python
wiith dask.config.set(collection_annotations={"blah": 1}:
   ...
```

I wonder if this should be documented, although I might say not if the intention is for ingestion by internal visualization routines.

Reply:
If we document it then we need to support it.  Right now all we need is for this to be supported by internal functions.  I don't see a reason to intentionally expose or advertise this out to users.  Given that this part of the codebase is under churn I'm inclined to not promise much to users here.

Reply:
> intention is for ingestion by internal visualization routines

I'd say this description is accurate.